source/timemory/hash/macros.hpp

```diff
@@ -41,3 +41,8 @@
 #else
 #    define TIMEMORY_HASH_LINKAGE(...) inline __VA_ARGS__
 #endif
+
+#if defined(_MSC_VER) && !defined(NDEBUG) && !defined(DEBUG)
+#    define DEBUG
+#endif
```

Comment:
So I failed to account for multi-config IDEs like Xcode and Visual Studio. You can remove this and remove:

```
        if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
            target_compile_definitions(${TARGET_NAME} PRIVATE DEBUG)
        endif()
```

from the `build_library` macro in `MacroUtilities.cmake`. Then add this at the top of BuildSettings.cmake:

```
target_compile_definitions(timemory-compile-options INTERFACE $<$<CONFIG:DEBUG>:DEBUG>)
```